ldap_validator/ldap_validator.py

```diff
@@ -0,0 +1,474 @@
+#!/usr/bin/env python3
+####################################################################
+# ### ldap_validator.py                                          ###
+####################################################################
+# ### Author: SAS Institute Inc.                                 ###
+####################################################################
+#                                                                ###
+# Copyright (c) 2020, SAS Institute Inc., Cary, NC, USA.         ###
+# All Rights Reserved.                                           ###
+#                                                                ###
+####################################################################
+
+import datetime
+from ldap3 import Server, Connection
+import os
+import sys
+import getopt
+import yaml
+import traceback
+import json
+import pprint
+import logging
+from ldap_validator.library.utils import ldap_messages
+from ldap3.core.exceptions import LDAPException
+from viya_ark_library.command import Command
+from viya_ark_library.logging import ViyaARKLogger
+try:
+    from yaml import CSafeLoader as Loader
+except ImportError:
+    from yaml import SafeLoader as Loader
+######################################################################################
+# retrieve at most sizelimit entries for a search.  A sizelimit of 0 (zero) or
+# none means no limit.  A sizelimit of  max  means  the maximum integer allowable
+# by the protocol.  A server may impose a maximal sizelimit which only the root
+# user may override.
+sizeLimit = 5
+
+# templates for output file names #
+_VALIDATOR_LOG_NAME_TMPL_ = "ldap_validator_{}.log"
+_FILE_TIMESTAMP_TMPL_ = "%Y-%m-%dT%H_%M_%S"
+
+# set timestamp for report file
+file_timestamp = datetime.datetime.now().strftime(_FILE_TIMESTAMP_TMPL_)
+
+
+# sitedefault variables
+ldap_server_host = ""
+ldap_server_port = ""
+ldap_protocol = ""
+ldap_anon_bind = ""
+ldap_bind_userdn = ""
+ldap_bind_pw = ""
+ldap_user_basedn = ""
+ldap_group_basedn = ""
+ldap_defaultadmin_user = ""
+ldap_url = ""
+
+
+# --------------------------------------------------------------------------------------------------
+def import_site_default(yaml_file, ldap_logger, sas_logger):
```

Comment:
What is the sas_logger input parameter here?

Reply:
sas_logger is passed to this function from main so that it could be accessed to get the full log file name/path. I think I need to remove the "global" setting for sas_logger at this point.

Reply:
You explained this to me in review meeting, so I understand the need now.   I was confused because I didn't see the same required methodology in pre-install-report.   There all of this is mitigated by the object-oriented implementation of pre-install-report.  

Don't change the "global" setting for sas_logger at this point since it all works as-is.